ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/exports/FeatureExportOperation.java

```diff
@@ -856,7 +857,11 @@ protected String[] getPaths() {
 		FeatureModelManager fmm = PDECore.getDefault().getFeatureModelManager();
 		IFeatureModel[] models = fmm.getExternalModels();
 		for (IFeatureModel model : models) {
-			map.put(model.getFeature().getId(), model.getInstallLocation());
+			if (model.getInstallLocation() != null) {
+				map.put(model.getFeature().getId(), model.getInstallLocation());
+			} else {
+				PDECore.log(Status.warning(NLS.bind(PDECoreMessages.FeatureExportOperation_featureWithoutInstallLocationFound, model.getFeature().getId())));
```

Comment:
At this stage, such error message is important for user, it's not a technical message that should be in log, but something the user should be warned about.
The unsupported features should be listed and then shown in a popup after export.
Or, even better, this should happen preventively in the UI by not allowing to select those features and instead mark them with the reason why they can't be selected.

Reply:
> Or, even better, this should happen preventively in the UI by not allowing to select those features and instead mark them with the reason why they can't be selected.

I don't think that's really the problem, given that both the plug-in and feature exporter only allow you to select projects in your workspace. So this case shouldn't be possible.

To be honest, I'm not even sure why it is then necessary to load those external features. 

And if you were to export something containing this synthetic feature, you end up with an error message, regardless. So an additional popup seems redundant.
![image](https://user-images.githubusercontent.com/92990747/196719653-fea14872-45f7-4631-ad0b-b5fda6e621ec.png)

Reply:
It looks like the change was made as part of [Bug 309884](https://bugs.eclipse.org/bugs/show_bug.cgi?id=309884). It makes sense that this feature would be required when exporting e.g. a product or the feature I've shown above. In which case it's the responsiblity of the external provider to set the install location, if they want it to work with the exporter.

The reason it shows also up for plug-ins is simply because all of those exporters use the same mechanism for creating the p2 structure and therefore it is just an unlucky side-effect. 

But if that's the case, then I agree that it doesn't make sense to log the error message. Instead, I would silently ignore those artifacts and instead rely on the pop-up that shows up when Eclipse tries to load the missing element.

Reply:
OK, it looks like not logging is fine. The rule is
* user needs to worry -> a popup
* some curiours developer or a contributor needs to worry -> a log message
* no-one needs to worry -> nothing.

Looks like there is an agreement it's the later case.